Adds the CPC (Clone Personnel Carrier) module

Lumipharon opened the pull request:
## About The Pull Request
Adds the PERFECTLY BALANCED, WELL THOUGHT OUT cloner module to the APC.
![image](https://github.com/tgstation/TerraGov-Marine-Corps/assets/7869430/9bc19bdb-349c-4cd0-bfff-fc84425b20e4)

You chug around, pumping out cheap clone troopers who can gear up with the noob vendor (or extra gear if you load it in).
Not pictured is the APC gibber, which produces additional biomass, so you can recycle bodies for YET MORE clone soldiers.
## Why It's Good For The Game
Marine spawner.
Balance perfection.
## Changelog
:cl:
add: Added a cloner module for the APC
/:cl:

Tonklaatech commented:
The weapon to surpass Metal Gear.

FosterGrime-Heymynameslouie commented:
spedemrhg instantly

74delta commented:
Holy fuck please

SimplySmiley123 commented:
So real.

iforgotmeword commented:
![IMG_0967](https://github.com/tgstation/TerraGov-Marine-Corps/assets/101154175/28ffc9ec-d4a0-4293-920f-c3835054870d)

same energy

ilikesaying commented:
robot version when

Admiral-Gigaton commented:
can you (if its not already implemented in this PR) make it so corpses can be loaded into the gibber from the outside, and/or gib xeno corpses for biomass (no one sends up minion corpses, so it would work out)

Wisemonster commented:
> can you, if its not already implemented in this PR, make it so corpses can be loaded into the gibber from the outside, and/or gib xeno corpses for biomass (no one sends up minion corpses, so it would work out)
> 
> its confusing and hard to load corpses n crates into an APC, and not many people know how to, so this would help. or you could make the info about loading things in more available to people.
> 
> otherwise, love this PR, hope it doesnt get shot down, and are there any other cons to doing this, besides getting shitty gear?
> 
> also, does this pull from ghost pool, or is it a DNA scanner moment. could i die, and become fully round removed, but my gear dropped onto the APC, where i then clone via ghost spawn, and pick up my shit?
> 
> QUESTIONS!!

To answer some of these, I'm pretty sure these are just the medbay cloners stuck into an apc, so they will work how the medbay cloner currently works

Admiral-Gigaton commented:
> > can you, if its not already implemented in this PR, make it so corpses can be loaded into the gibber from the outside, and/or gib xeno corpses for biomass (no one sends up minion corpses, so it would work out)
> > its confusing and hard to load corpses n crates into an APC, and not many people know how to, so this would help. or you could make the info about loading things in more available to people.
> > otherwise, love this PR, hope it doesnt get shot down, and are there any other cons to doing this, besides getting shitty gear?
> > also, does this pull from ghost pool, or is it a DNA scanner moment. could i die, and become fully round removed, but my gear dropped onto the APC, where i then clone via ghost spawn, and pick up my shit?
> > QUESTIONS!!
> 
> To answer some of these, I'm pretty sure these are just the medbay cloners stuck into an apc, so they will work how the medbay cloner currently works

I have no idea how the medbay cloner works, i havent been cloned before (besides 1 case but i barely remember)

bowlofcereal commented:
this better start with a lot of biomass considering it takes like one whole beaker of biomass for one clone and there are 8 of these, so 8 biomass beakers each time it creates one

i've got a feeling this is going to be a module pick that makes people scream "griefer" because of this

lbnesquik commented:
> this better start with a lot of biomass considering it takes like one whole beaker of biomass for one clone and there are 8 of these, so 8 biomass beakers each time it creates one
> 
> i've got a feeling this is going to be a module pick that makes people scream "griefer" because of this

Bowl is right, biomass is expensive to run.

Lumipharon commented:
>they don't know

00ze-cyclone commented:
> > this better start with a lot of biomass considering it takes like one whole beaker of biomass for one clone and there are 8 of these, so 8 biomass beakers each time it creates one
> > i've got a feeling this is going to be a module pick that makes people scream "griefer" because of this
> 
> Bowl is right, biomass is expensive to run.

Biomass is expensive sure but you can get more biomass, you can’t get more cloners, those are the true bottleneck of the clone army, then it’s just a matter of fundings

lbnesquik commented:
They did mention a gibber for biomass. It's just, you can't drag xeno bodies inside the APC, can you?

TiviPlus commented on code/game/objects/machinery/cloning/cloning.dm:
```diff
@@ -68,7 +68,7 @@ These act as a respawn mechanic growning a body and offering it up to ghosts.
 
 	if(!linked_machine)
 		// Try to find the machine nearby
-		linked_machine = locate() in orange(1, src)
+		linked_machine = locate() in get_step(src, REVERSE_DIR(dir))
```
feels like we should check this in init as well and have a log_mapping fail

yyzsong commented:
> this better start with a lot of biomass considering it takes like one whole beaker of biomass for one clone and there are 8 of these, so 8 biomass beakers each time it creates one
> 
> i've got a feeling this is going to be a module pick that makes people scream "griefer" because of this

you get like two and a half clones per beaker you'll be fine

Looking4bros commented:
I'm pretty sure you get 3 clones per 2 beakers, 40u per clone as they give you regular beakers and not large beakers.

Wisemonster commented:
Might be a good idea to tell where each clone is, cause currently it's not possible to tell which clone is in the apc, and which is in medbay

Looking4bros commented:
This is 'uge for Vali, instead of waiting 5-20 minutes getting surgery for necro, or lost limbs, return you body to the grinder, and take a damage penalty for having 0 melee skill for a while. It's Vali, you're not really damaging things anyway.

74delta commented:
This is incredibly based. 10/10

zee11zee commented:
I'm repeating here "Baneling but better"

Kuro020 reviewed (changes requested):
![image](https://github.com/tgstation/TerraGov-Marine-Corps/assets/52428261/2f1d1d91-3152-4b22-820a-1d102cdac5aa)
Really brother

TiviPlus commented:
fix the clone turbo spam issue pls, this module outclasses the other two by far

Lauwma commented:
Either it's baneling for marines or a point generator for xenos. I'd say close the PR. if marines aren't doing well with their custom finely tunned loadout what makes you think they will perform as fragile clones with generic loadouts.

Looking4bros commented:
Perhaps... in species_types/vatborns.dm you could make a third Vatborn APC variant and just make it absolute dogwater, Because this gimmick is very funny, and perhaps would be fine if the clones were just soooo bad. All zeros in skills or a larger damage modifier applied to them without progression to full marine strength.

Also, the gibber can be interacted with as a ghost, and it ejects the beaker.

Lumipharon commented:
> fix the clone turbo spam issue pls, this module outclasses the other two by far

What do you want me to change? I made it because it was funny, not with serious balance in mind, but I'm open to change whatever.
Less cloners? Longer clone time? less biomass per body?

74delta commented:
Turn it into a robot version and make it take mats to make instead of biomass so it's more expensive and less common 🧠

chizzy376 commented:
After 5 rounds of using this, I can definitely say that this is the best module for preventing losing the game due to marine life attrition
Some map spawns have a lot of generous corpses and you can pump out a bunch of clones
Only request is that you should be able to click the console on a finished growth vat to re-alert ghosts that the clone is available for taking (on a cooldown) so you don't have to give hints to ghosts over radio

TiviPlus commented:
> > fix the clone turbo spam issue pls, this module outclasses the other two by far
> 
> What do you want me to change? I made it because it was funny, not with serious balance in mind, but I'm open to change whatever. Less cloners? Longer clone time? less biomass per body?

iirc main issue is that people scrape up the prespawn bodies to get endless clones

Pockets-byte commented:
have people put revivable bodies into the gibber? because that sounds like that's something people would do

Lumipharon commented:
> > > fix the clone turbo spam issue pls, this module outclasses the other two by far
> > 
> > 
> > What do you want me to change? I made it because it was funny, not with serious balance in mind, but I'm open to change whatever. Less cloners? Longer clone time? less biomass per body?
> 
> iirc main issue is that people scrape up the prespawn bodies to get endless clones

Changed so mapspawn bodies can no longer be gibbed for biomass.
Changed the base biomass per meat from 5 to 1, so a lot less biomass over all.

Also fixed some vehicle runtimes while I was at it.

ivanmixo commented:
Don't think adding marine baneling is a good idea, even if it is funny

chizzy376 commented:
> Changed so mapspawn bodies can no longer be gibbed for biomass

Good change! But I'd like to know,

> Changed the base biomass per meat from 5 to 1, so a lot less biomass over all.

Does this mean that instead of needing to gib 2 DNR marines, you need to gib 10 to make enough biomass?

TiviPlus commented:
Looks to be okish now?
